src/edu/asu/plp/tool/prototype/Main.java

```diff
@@ -413,45 +463,843 @@ private ProjectExplorerTree createProjectTree()
+				directoryChooser.setTitle("Choose Project Location");
+				File file = directoryChooser.showDialog(null);
+				// If Cancel is chosen, throws a null pointer, needs to be fixed
+				chosenLocation = file.getAbsolutePath();
+				projLocationField.setText(chosenLocation);
+			}
+		});
+		
+		Label target = new Label();
+		target.setText("Targetted ISA: ");
+		target.setFont(Font.font("Arial", FontWeight.NORMAL, 16));
+		
+		ComboBox<String> isaType = new ComboBox<String>();
+		isaType.getItems().addAll("PLP", "MIPS");
+		isaType.setValue("PLP");
+		
+		Label version = new Label();
+		version.setText("Version: ");
+		version.setFont(Font.font("Arial", FontWeight.NORMAL, 16));
+		
+		grid.add(projectName, 0, 0);
+		grid.add(projTextField, 1, 0);
+		grid.add(mainSourceFile, 0, 1);
+		grid.add(sourceFileField, 1, 1);
+		grid.add(projectLocation, 0, 2);
+		grid.add(projLocationField, 1, 2);
+		grid.add(browseLocation, 2, 2);
+		grid.add(target, 0, 3);
+		grid.add(isaType, 1, 3);
+		grid.add(version, 0, 4);
+		
+		border.setCenter(grid);
+		
+		Button createProject = new Button("Create Project");
+		createProject.setOnAction(new EventHandler<ActionEvent>() {
+			@Override
+			public void handle(ActionEvent e)
+			{
+				String projectName;
+				String fileName;
+				String projectLocation;
+				Alert alert = new Alert(AlertType.INFORMATION);
+				projectName = projTextField.getText();
+				fileName = sourceFileField.getText();
+				projectLocation = projLocationField.getText();
+				if (projectName.equals(""))
+				{
+					alert.setTitle("Invalid Project Name");
+					alert.setHeaderText(null);
+					alert.setContentText("You entered and invalid Project Name");
+					alert.showAndWait();
+					
+				}
+				else if (fileName.equals(""))
+				{
+					alert.setTitle("Invalid Project Name");
+					alert.setHeaderText(null);
+					alert.setContentText("You entered and invalid File Name");
+					alert.showAndWait();
+				}
+				else if(projectLocation.equals(""))
+				{
+					alert.setTitle("Invalid Project Loaction");
+					alert.setHeaderText(null);
+					alert.setContentText("You entered and invalid Project Locaction");
+					alert.showAndWait();
+					
+				}
+				else
+				{
+					projectName = projTextField.getText();
+					fileName = sourceFileField.getText();
+					
+					System.out.println(fileName);
+					
+					if (!fileName.contains(".asm"))
+					{
+						fileName = fileName.concat(".asm");
+					}
+					System.out.println("After: " + fileName);
+					
+					CodeEditor content = createCodeEditor();
+					content.setText("#New PLP Project");
+					PLPProject project = new PLPProject(projectName);
+					project.setPath(projLocationField.getText());
+					PLPSourceFile sourceFile = new PLPSourceFile(project, fileName);
+					project.add(sourceFile);
+					project.save();
+					projects.add(project);
+					openFile(sourceFile);
+					Stage stage = (Stage) createProject.getScene().getWindow();
+					stage.close();
+				}
+			}
+		});
```

Comment:
Perhaps this window is substantial enough to warrant its own class...
We should discuss this when we talk about refactoring (this can be put off until next sprint)